### ISSUE TITLE
Bitlbee Restarting on Account Activation

### ISSUE THREAD

**Issue by aclindsa** [hoehermann/purple-signald#32]
I'm attempting to get this plugin working with bitlbee on Arch Linux. I've built the latest master (https://github.com/hoehermann/libpurple-signald/commit/90cc584f25a4d3b449913d9b963e67ace4dfa4d4) and am using signald 0.11.0 (via https://aur.archlinux.org/packages/signald/). I'm able to send-receive messages via signald by talking to the socket directly with netcat - so that portion seems to be working okay.

And it appears that bitlbee is successfully recognizing libpurple-signald. I am able to successfully do:
```
add hehoe-signald +12223334444
```
(though I had to do `account add hehoe-signald...` instead, and obviously used my own phone number). I can later see with `account list` that it has created the account. However, the next step (rename _12223334444 name-sig) does not work for be because no buddies are registered. Finally (and I think this is the real bug), when I do `account  hehoe-signald on`, bitlbee unceremoniously restarts itself. When I peek in the bitlbee log from this time period, I see the following messages:

> Jan 19 15:12:10 host bitlbee[4036611]: purple_buddy_new: assertion 'name != NULL' failed
> Jan 19 15:12:10 host bitlbee[4036611]: purple_blist_add_buddy: assertion 'buddy != NULL' failed
> Jan 19 15:12:10 host bitlbee[4036611]: purple_blist_alias_buddy: assertion 'buddy != NULL' failed

I'm happy to help debug this further if you point me in the right direction.

**Comment by hoehermann** [comment]
I have a hunch this may be a side effect of the new groups and profiles. Buddies can now be identified by an UUID rather than their number. In this case, the number can be NULL. That indeed would make it impossible to add the buddy. It should not make bitlbee crash, though.

Or it has something to to with the buddy list download. Maybe @herm can look into this. I am not using that particular feature nor bitlbee so I we are dependant on developers who do.

**Comment by Strykar** [comment]
I can confirm that buddies in my groups do not show phone numbers anymore, but Bitlbee and signal are working fine for me on Arch linux. I had set up libpurple-signald before the V2 / groups change though.

**Comment by aclindsa** [comment]
@hoehermann @Strykar Is there a good way for me to add more verbose logging to libpurple-signald through bitlbee? For example, I'm wondering if seeing the messages between libpurple-signald <-> signald interspersed with the assertions I printed above would help narrow down the issue further.

If not, is there a best practice for gaining more information about what's going on - is my best bet to run bitlbee in gdb when I get a chance and see where that leads?

**Comment by hoehermann** [comment]
@aclindsa As far as I can tell, bitlbee does not crash, but it rather chokes on some input. Using gdb probably won't help much. The plug-in prints each incoming line (JSON) using `purple_debug_info` [here](
https://github.com/hoehermann/libpurple-signald/blob/86dc0cd6076ef7ff88a47052d8a81f74c4a0317f/libsignald.c#L254). Unfortunately, I do not know where bitlbee puts that information.

**Comment by aclindsa** [comment]
Okay, this is really weird - I figured out how to enable debug logging in bitlbee this morning to see if it would pass through those messages, started it up using the exact same thing I did a million times yesterday and.... it's working perfectly fine. I do still see some assertions, but they're not the same ones I was seeing yesterday. And in case its not clear, I hadn't actually enabled debugging yet or touched my config in any way since yesterday.

Given that it's working now, I'm going to close this for now, and will plan to re-open if I encounter the issue again and am able to provide more information.
